extra_sdk_options.go

```diff
@@ -146,13 +162,74 @@ func NewWithCredentials(ctx context.Context, cred *ClientCredentials, opts ...Cu
 	}
 
 	sdkOpts := []SDKOption{}
-	if options.useWithServer {
-		sdkOpts = append(sdkOpts, WithServerURL(options.ServerURL))
+	if options.UseWithServer() {
+		sdkOpts = append(sdkOpts, WithServerURL(options.ServerURL()))
 	}
-	if options.useWithTenant {
-		sdkOpts = append(sdkOpts, WithTenantDomain(options.Tenant))
+	if options.UseWithTenant() {
+		sdkOpts = append(sdkOpts, WithTenantDomain(options.Tenant()))
 	}
 	sdkOpts = append(sdkOpts, WithClient(uclient))
 
 	return New(sdkOpts...), nil
 }
+
+func NormalizeTenant(input string) (*ClientConfig, error) {
+	input = strings.ToLower(input)
+
+	var err error
+	u := &url.URL{}
+	if !strings.Contains(input, "//") {
```

Comment:
Just curious, but why not try to parse the string as a URL first, then fall back to adding the c1 tenant domain?

Also if you leave the code this way, I think you could check for `"://"` to reduce the risk of false positives.

Reply:
Added the :// change. 

The url.Parse method does say "Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities." It is hard to reason through what happens in that case, I think explicitly handling it first maybe better but I am open to revisiting this in another ticket